src/Agent.Worker/StepsRunner.cs

```diff
@@ -295,6 +298,7 @@ private async Task RunStepAsync(IStep step, CancellationToken jobCancellationTok
             }
 
             // Complete the step context.
+            step.ExecutionContext.SetStepTarget(null);
```

Comment:
I wonder if a `using` pattern would make this read better. Assuming we created a disposable class that called SetStepTarget on the context correctly...
```cs
using (var stepTargetManager = new StepTargetManager(step))
{ ... }
```

Reply:
Setting null actually isn't necessary since each step gets its own `ExecutionContext`. I'll remove that line instead.